src/Microsoft.IdentityModel.Tokens/Json/EncodedJsonWebKeyParameterNames.cs

```diff
@@ -0,0 +1,271 @@
+                if (!_pSet)
+                {
+                    _p = JsonEncodedText.Encode(JsonWebKeyParameterNames.P, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _pSet = true;
+                }
+
+                return _p;
+            }
+        }
+        public static JsonEncodedText Q
+        {
+            get
+            {
+                if (!_qSet)
+                {
+                    _q = JsonEncodedText.Encode(JsonWebKeyParameterNames.Q, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _qSet = true;
+                }
+
+                return _q;
+            }
+        }
+
+        public static JsonEncodedText QI
+        {
+            get
+            {
+                if (!_qiSet)
+                {
+                    _qi = JsonEncodedText.Encode(JsonWebKeyParameterNames.QI, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _qiSet = true;
+                }
+
+                return _qi;
+            }
+        }
+
+        public static readonly JsonEncodedText Use = JsonEncodedText.Encode(JsonWebKeyParameterNames.Use, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+
+        public static readonly JsonEncodedText X5c = JsonEncodedText.Encode(JsonWebKeyParameterNames.X5c, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+
+        public static readonly JsonEncodedText X5t = JsonEncodedText.Encode(JsonWebKeyParameterNames.X5t, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+
+        public static JsonEncodedText X5tS256
+        {
+            get
+            {
+                if (!_x5tS256Set)
+                {
+                    _x5tS256 = JsonEncodedText.Encode(JsonWebKeyParameterNames.X5tS256, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _x5tS256Set = true;
+                }
+
+                return _x5tS256;
+            }
+        }
+
+        public static JsonEncodedText X5u
+        {
+            get
+            {
+                if (!_x5uSet)
+                {
+                    _x5u = JsonEncodedText.Encode(JsonWebKeyParameterNames.X5u, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _x5uSet = true;
+                }
+
+                return _x5u;
+            }
+        }
+        
+        public static JsonEncodedText X
+        {
+            get
+            {
+                if (!_xSet)
+                {
+                    _x = JsonEncodedText.Encode(JsonWebKeyParameterNames.X, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _xSet = true;
+                }
+
+                return _x;
+            }
+        }
+
+        public static JsonEncodedText Y
+        {
+            get
+            {
+                if (!_ySet)
+                {
+                    _y = JsonEncodedText.Encode(JsonWebKeyParameterNames.Y, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
+                    _ySet = true;
+                }
+
+                return _y;
+            }
```

Comment:
You could simplify this file with a helper like:
```C#
private static JsonEncodedText Get(ref bool initialized, ref JsonEncodedText field, string name)
{
    if (!Volatile.Read(ref initialized)
    {
        field = JsonEncodedText.Encode(name, JavaScriptEncoder.UnsafeRelaxedJsonEscaping);
        Volatile.Write(ref initialized, true);
    }
    return field;
}
```
and then your properties all become one-liners, e.g.
```C#
public static JsonEncodedText Y => Get(ref _ySet, ref _y, JsonWebKeyParameterNames.Y);
```
(and then you don't need to make the bool fields volatile as it's then handled by the Get helper).